### PR TITLE
reuse code

### PR DESCRIPTION
**Pull request by individual-it**
reuse code for adding scripts/styles

### CODE REVIEW

File: lib/private/util.php [hunk 1]
```diff
@@ -535,13 +507,36 @@ public static function addStyle($application, $file = null, $prepend = false) {
...
+	/**
+	 * add an external ressourse css/js file
+	 *
+	 * @param string $application application id
+	 * @param bool $prepend prepend the file to the beginning of the list
+	 * @param string $path 
+	 * @param string $type (script or style)
+	 * @return void
+	 */
+	private static function addExternalRessource($application, $prepend, $path, $type = "script") {
```
Comment (phil-davis) [inline lib/private/util.php]: In function name and comments - "Ressource" or "ressourse" change to "Resource"

File: lib/private/util.php [hunk 2]
```diff
@@ -535,13 +507,36 @@ public static function addStyle($application, $file = null, $prepend = false) {
...
+	 *
+	 * @param string $application application id
+	 * @param bool $prepend prepend the file to the beginning of the list
+	 * @param string $path 
+	 * @param string $type (script or style)
+	 * @return void
+	 */
+	private static function addExternalResource($application, $prepend, $path, $type = "script") {
+
+		if ($type == "style") {
```
Comment (MorrisJobke) [inline lib/private/util.php]: Please use `===` instead of `==`

File: lib/private/util.php [hunk 3]
```diff
@@ -535,13 +507,36 @@ public static function addStyle($application, $file = null, $prepend = false) {
...
+			if (!in_array($path, self::$styles)) {
+				if ($prepend === true) {
+					array_unshift ( self::$styles, $path );
+				} else {
+					self::$styles[] = $path;
+				}
+			}
+		} elseif ($type == "script") {
```
Comment (MorrisJobke) [inline lib/private/util.php]: Same here

**Comment by MorrisJobke** [comment]
cc @PVince81 @nickvergessen

**Comment by MorrisJobke** [comment]
Generally speaking I would like to see this covered by a unit test and migrated to non-static code.

**Comment by MorrisJobke** [comment]
:see_no_evil: :hear_no_evil: :speak_no_evil:

**Comment by MorrisJobke** [comment]
cc @DeepDiver1975

**Comment by PVince81** [comment]
It should be possible to unit test this even though it's static.
The `tearDown` would have to clear the `self::$scripts` variable after each run (we have a method `invokePrivate` that can achieve this)

**Comment by MorrisJobke** [comment]
I added unit tests for this. 


It is now ready for review :) cc @nickvergessen @rullzer @PVince81

**Comment by MorrisJobke** [comment]
I also tested it and it still works :)

**Comment by MorrisJobke** [comment]
I also ran the unit tests without the patches in here to verify they don't cover only the "new" behavior. ;)

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by MorrisJobke** [comment]
As I only added unit tests: Here is my :+1: for the actual change

**Pull request merged**